collect_app/src/main/java/org/odk/collect/android/utilities/ThemeUtils.java

```diff
@@ -27,18 +27,22 @@
 import org.odk.collect.android.preferences.GeneralSharedPreferences;
 import org.odk.collect.android.preferences.PreferenceKeys;
 
+import javax.inject.Inject;
+
 public final class ThemeUtils {
 
-    private final Context context;
-    private final GeneralSharedPreferences generalSharedPreferences;
+    @Inject
+    GeneralSharedPreferences generalSharedPreferences;
+    @Inject
+    Context context;
```

Comment:
I don't see where this `context` ia being utilized. I see the `context` being passed for each individual method that needs it so should this be removed?

Reply:
It is being used [here](https://github.com/shobhitagarwal1612/collect/blob/b426d5d97e7302a8064605e10f922c98d1e01341/collect_app/src/main/java/org/odk/collect/android/utilities/ThemeUtils.java#L45)

The reason I had to do this was that whenever an activity is opened, `ThemeUtils` checks the current saved theme in the preferences and then applies the value for which a `context` was required. Since the preference has to be queried so the Application's context can work as well (which is provided by the above `@Inject`). But when the attributes from the currently loaded theme have to be fetched, then only an activity's instance can be used.

Reply:
Okay understood. Makes sense.